LibraryInstrumentationGuidance.md

```diff
@@ -0,0 +1,107 @@
+# Guidence for instrumenting libraries with Diagnostic Source 
+
+This document provides guidence for adding Diagnostic Source instrumentation to external libraries, which will allow Application Insights and other monitoring solutions to collect meaningful and rich telemetry.
+
+## Diagnostic Source and Activities
+
+[Diagnostic Source][DiagnosticSourceGuide] is a simple module that allows code to be instrumented for production-time logging of rich data payloads for consumption within the process that was instrumented. At runtime consumers can dynamically discover data sources and subscribe to the ones of interest.
+
+[Activity][ActivityGuide] is a class that allows storing and accessing diagnostics context and consuming it with logging system.
+
+Both Diagnostic Source and Activity have been used to instrument [System.Net.Http][SystemNetHttp] and [Microsoft.AspNetCore.Hosting][MicrosoftAspNetCoreHosting], although that instrumentation is not fully complaiant with this guidance.
```

Comment:
Referring SystemNetHttp and AspNetCore.Hosting is good as it builds trust with this guidance. And then the next sentence confuses people. Are they using it incorrectly? Are they ignoring the part of guidance which might hurt performance?

The point I'm making - do we need to mention this at all? How this information should be used by readers of this guidance?

Reply:
Good point, I've removed that remark